src/services/itemMembership/repository.ts

```diff
@@ -93,6 +96,75 @@ export const ItemMembershipRepository = AppDataSource.getRepository(ItemMembersh
     return query.getMany();
   },
 
+  /**
+   *  get accessible items for actor and given params
+   *  */
+  async getAccessibleItems(
+    actor: Member,
+    { creatorId, name, sortBy = SortBy.ItemUpdatedAt, ordering = Ordering.desc }: ItemSearchParams,
+    { page = 1, pageSize = ITEMS_PAGE_SIZE }: PaginationParams,
+  ): Promise<Paginated<Item>> {
+    const limit = Math.min(pageSize, ITEMS_PAGE_SIZE_MAX);
+    const skip = (page - 1) * limit;
+
+    const query = this.createQueryBuilder('im')
+      .leftJoinAndSelect('im.item', 'item')
+      .leftJoinAndSelect('item.creator', 'creator')
+      .where('im.member_id = :actorId', { actorId: actor.id })
+      // returns only top most item
+      .andWhere((qb) => {
+        const subQuery = qb
+          .subQuery()
+          .from(ItemMembership, 'im1')
+          .select('im1.item.path')
+          .where('im.item_path <@ im1.item_path')
+          .orderBy('im1.item_path', 'ASC')
+          .limit(1)
+          .getQuery();
+        return 'item.path =' + subQuery;
+      });
+
+    if (name) {
+      query.andWhere("LOWER(item.name) LIKE '%' || :name || '%'", {
+        name: name.toLowerCase().trim(),
+      });
+    }
+
+    if (creatorId) {
+      query.andWhere('item.creator = :creatorId', { creatorId });
+    }
+
+    if (sortBy) {
+      // map strings to correct sort by column
+      let mappedSortBy;
+      switch (sortBy) {
+        case SortBy.ItemType:
+          mappedSortBy = 'item.type';
+          break;
+        case SortBy.ItemUpdatedAt:
+          mappedSortBy = 'item.updated_at';
+          break;
+        case SortBy.ItemCreatedAt:
+          mappedSortBy = 'item.created_at';
+          break;
+        case SortBy.ItemCreatorName:
+          mappedSortBy = 'creator.name';
+          break;
+        case SortBy.ItemName:
+          mappedSortBy = 'item.name';
+          break;
```

Comment:
Factor out to (local/util) function ?

Reply:
I wonder too, I think it's better than each function implement their mapper? Also it may be easier to see if there's a case missing? idk